### PR TITLE
Creating content for Troubleshooting section in the GitOps 1.8 Release

### PR DESCRIPTION
**Pull request by Dhruv-Soni11**
Purpose: To resolve the following issue:

https://issues.redhat.com/browse/RHDEVDOCS-3825

Aligned team: DevTools

OCP version this PR applies to (cherrypicking): enterprise 4.11 and later

Content for preview: https://56809--docspreview.netlify.app/openshift-enterprise/latest/cicd/gitops/preventing-auto-reboot-during-argo-cd-sync-with-machine-configurations.html

SME review : @iam-veeramalla   

QE review: @varshab1210 

Peer review: @themr0c

### CODE REVIEW

**Comment by varshab1210** [comment]
LGTM. Thanks

File: con_performance-challenges-in-machine-configurations-and-argo-cd.adoc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+:_content-type: CONCEPT
+
+[id="performance-challenges-in-machine-configurations-and-argo-cd_{context}"]
+= Performance challenges in Machine configurations and Argo CD
+include::_attributes/common-attributes.adoc[]
+:context: performance-challenges-in-machine-configurations-and-argo-cd
+
+When you are using a Machine Config Operator as part of a GitOps workflow, the following sequence can produce suboptimal performance:
+
+* Argo CD starts a automated sync job after a commit to the Git repository that contains application resources.
```
Comment (themr0c) [inline con_performance-challenges-in-machine-configurations-and-argo-cd.adoc:10]:
```suggestion
* Argo CD starts an automated sync job after a commit to the Git repository that contains application resources.
```

File: cicd/gitops/preventing-auto-reboot-during-argo-cd-sync-with-machine-configurations.adoc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+:_content-type: ASSEMBLY
+[id="preventing-auto-reboot-during-argo-cd-sync-with-machine-configurations"]
+
+= Preventing auto-reboot during Argo CD sync with Machine Configurations
+include::_attributes/common-attributes.adoc[]
+:context: preventing-auto-reboot-during-argo-cd-sync-with-machine-configurations
+
+In the Red Hat OpenShift Container Platform, nodes are updated automatically through the Red Hat OpenShift Machine Config Operator (MCO). A Machine Config Operator (MCO) is a custom resource that is used by the cluster to manage the complete life cycle of its nodes. When an MCO resource is created or updated in a cluster, the MCO picks up the update, performs the necessary changes to the selected nodes, and restarts the nodes gracefully by cordoning, draining, and rebooting those nodes. It handles everything from the kernel to the kubelet.
```
Comment (themr0c) [inline cicd/gitops/preventing-auto-reboot-during-argo-cd-sync-with-machine-configurations.adoc:8]: I assume you don't do one sentence per line. https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line

File: modules/performance-challenges-in-machine-configurations-and-argo-cd.adoc [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+[id="performance-challenges-in-machine-configurations-and-argo-cd"]
+= Performance challenges in Machine configurations and Argo CD
+
+When you are using a Machine Config Operator as part of a GitOps workflow, the following sequence can produce suboptimal performance:
+
+* Argo CD starts a automated sync job after a commit to the Git repository that contains application resources.
+
+* If Argo CD notices a new or an updated machine configuration while the sync operation is in process, MCO picks up the change to the machine configuration and starts rebooting the nodes to apply the change.
+
+* If a rebooting node in cluster contains the Argo CD application controller, the application controller terminates and the application sync is aborted.
```
Comment (themr0c) [inline modules/performance-challenges-in-machine-configurations-and-argo-cd.adoc:12]:
```suggestion
* If a rebooting node in the cluster contains the Argo CD application controller, the application controller terminates and the application sync is aborted.
```

File: con_performance-challenges-in-machine-configurations-and-argo-cd.adoc [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+* If Argo CD notices a new or an updated machine configuration while the sync operation is in process, MCO picks up the change to the machine configuration and starts rebooting the nodes to apply the change.
+
+* If a rebooting node in cluster contains the Argo CD application controller, the application controller terminates and the application sync is aborted.
```
Comment (themr0c) [inline con_performance-challenges-in-machine-configurations-and-argo-cd.adoc:14]:
```suggestion
* If a rebooting node in cluster contains the Argo CD application controller, the application controller terminates, and the application sync is aborted.
```

**Comment by themr0c** [comment]
Good, except for some nitpicks.

**Comment by shipsing** [comment]
@Dhruv-Soni11 i want to give a suggestion on a broad level. please change the assembly name to something generic so that we can cover a wider range of topis into it in future. Can we change the assembly name to something like: `Troubleshooting GitOps issues`.
If you keep the current title `Preventing auto-reboot during Argo CD sync with Machine Configurations`, you make this assembly very specific, leaving out the scope for any future additions into this assembly. An assembly is usually a collection of multiple modules. If you keep the current assembly title, then we would not be able to add anything else into this assembly later, making the assembly stagnant for future.

File: con_performance-challenges-in-machine-configurations-and-argo-cd.adoc (hunk 1 above)
Comment (Dhruv-Soni11) [reply con_performance-challenges-in-machine-configurations-and-argo-cd.adoc:10]: Fixed.

File: modules/performance-challenges-in-machine-configurations-and-argo-cd.adoc (hunk 3 above)
Comment (Dhruv-Soni11) [reply modules/performance-challenges-in-machine-configurations-and-argo-cd.adoc:12]: Fixed.

File: con_performance-challenges-in-machine-configurations-and-argo-cd.adoc (hunk 4 above)
Comment (Dhruv-Soni11) [reply con_performance-challenges-in-machine-configurations-and-argo-cd.adoc:14]: Fixed.

**Comment by Dhruv-Soni11** [comment]
Hey @themr0c, I have incorporated the comments you provided for the comment. I have also made slight adjustments based on the suggestions provided by @shipsing which I feel increase the value of the content as well. Could you please have a look and provide your LGTM if everything now is as per your expectations?

**Comment by shipsing** [comment]
> @Dhruv-Soni11 i want to give a suggestion on a broad level. please change the assembly name to something generic so that we can cover a wider range of topis into it in future. Can we change the assembly name to something like: `Troubleshooting GitOps issues`. If you keep the current title `Preventing auto-reboot during Argo CD sync with Machine Configurations`, you make this assembly very specific, leaving out the scope for any future additions into this assembly. An assembly is usually a collection of multiple modules. If you keep the current assembly title, then we would not be able to add anything else into this assembly later, making the assembly stagnant for future.

Thank you @Dhruv-Soni11  for looking into this comment

**Review by themr0c — APPROVED** [review]

**Pull request merged**